crates/nu-parser/src/parser.rs

```diff
@@ -1425,6 +1425,21 @@ pub fn parse_range(working_set: &mut StateWorkingSet, span: Span) -> Expression
             return garbage(span);
         }
     };
+    // Avoid calling sub-parsers on unmatched parens, to prevent quadratic time on things like ((((1..2))))
+    // No need to call the expensive parse_value on "((((1"
+    if dotdot_pos[0] > 0 {
+        let (_tokens, err) = lex(
```

Comment:
Am I reading this correctly that we're lexing an additional time to do the check of whether or not it's valid before we proceed?

I wonder if there's some way to reuse what we learn.

Reply:
Yes, there is definitely a way to reuse what we learn, I mainly wasn't sure how to integrate with the fact that the lexer has parameters and different parts of the parser wants different parameters. I think a better solution overall would be for the lexer to return all the parentheses it parsed instead of tossing that info away, so we can reduce the number of times we need to re-lex the whole thing in the parser code in general.